test/functional/plugin/lsp/inline_completion_spec.lua

```diff
@@ -183,6 +184,57 @@ describe('vim.lsp.inline_completion', function()
       feed('<Esc>')
       screen:expect({ grid = grid_applied_candidates })
     end)
+
+    it('accepts on_accept callback', function()
+      feed('i')
+      screen:expect({ grid = grid_with_candidates })
+      local result = exec_lua(function()
+        ---@type vim.lsp.inline_completion.Item
+        local result
+        vim.lsp.inline_completion.get({
+          on_accept = function(item)
+            result = item
+          end,
+        })
+        vim.wait(100) -- Wait for async callback
```

Comment:
`vim.wait()` takes a callback so it can check `result` in 10 ms increments. Would that avoid the hardcoded 100 ms wait, and a potentially flaky test?

Reply:
Or we can use the `retry` utility: https://github.com/neovim/neovim/blob/a33284c2c068a9204330bc941e274b1f5bd9f5c6/test/testutil.lua#L75